ls1intum/Artemis
# `Text Exercises`: Link TextBlock to Feedback with feedback_id column

### Checklist
#### General


#### Server

### Motivation and Context


In order to suggest an automatic feedback from existing text blocks of old assessments, we need to keep a reference of the feedback id to the text block. This id will help us to suggest a feedback for all text blocks which belong to the same cluster as the existing ones.

### Description

Whenever we need to save the text blocks on the database, we need to set the Feedback based on the Feedback which has the reference equal to the corresponding Text Block. There is a foreign key `ON DELETE SET NULL` which prevents keeping a reference to a deleted text block. If the text blocks already exist on the repository, we only update the feedback_id value to the new created Feedback element!

### Steps for Testing

Prerequisites:
- 1 Instructor
- 2 Students
- 1 or more Text Exercises

1. Log in to Artemis
2. Navigate to Course Administration
3. Create and Import Text and Modeling exercises
4. Participate in the above-created exercises
5. Assess exercises
6. Delete Result
7. Submit a new submission
8. Assess the new submission
9. Check if any errors occur during different scenarios
10. If possible check if `feedback_id` value is set correctly on the database

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage
